cts/cts-fencing.in

```diff
@@ -579,13 +579,12 @@ class FenceTests(Tests):
 
         # use a host name that won't be in corosync.conf
         test.add_cmd("stonith_admin",
-                     args='--output-as=xml -R true1 -a fence_dummy -o mode=pass -o pcmk_host_list=regr-test')
+                     args='--output-as=xml -R true1 -a fence_dummy_no_nodeid '
```

Comment:
In "standalone" mode, the fencer always set `scheduler->priv->local_node_name` to localhost, even with corosync

Reply:
With corosync, `stand_alone` was set to `FALSE` and `options.no_cib_connect=true` -- see `stand_alone_cpg_cb()`. The local node was set to `"localhost"` only if `stand_alone` was `TRUE`.

Did the test actually break, or was this meant to be proactive?

Reply:
It broke. Note the absence of "True" at the end of the call previously -- that means it was run only in non-clustered standalone mode. Thus the local node was "localhost" and had no id. Now that we're running it with corosync active, it would send the local node's id with the request (since fence_dummy did support nodeid).

Reply:
> It broke.

I'm glad this fixes it.

> absence of "True"

As far as I can see, the `True` (for `with_cpg` argument) goes in the `new_test()` call (not the `add_cmd()` call) and it was present:
```
       ### verify nodeid is _NOT_ supplied when nodeid is not in the metadata parameters
        test = self.new_test("cpg_do_not_supply_nodeid",
                             "Verify nodeid is _NOT_ given when fence agent does not have nodeid as parameter",
                             True)

        # use a host name that won't be in corosync.conf
        test.add_cmd("stonith_admin",
                     args='--output-as=xml -R true1 -a fence_dummy -o mode=pass -o pcmk_host_list=regr-test')
```

Reply:
Ah right. I can't remember now how the name actually became localhost :) but in any case this implements the intent of the test and did fix it